LPS-138446 Turn off all of the new aui eslint rules

kresimir-coko opened the pull request:
Hey guys, this PR disables the eslint rules added in https://github.com/liferay/liferay-frontend-projects/pull/620. The rules are enabled and throw errors by default, so this PR will target them and disable them so that we can enable them one at a time.

This PR doesn't depend on, but does nothing without the version of `eslint-config` and `npm-scripts` that are still to be published and used in DXP.

I preemptively put this as a draft as I'm not sure what the sequence of event we are gonna want with these PRs.

/cc @bryceosterhaus

liferay-continuous-integration commented:
<h1>Build started.</h1><p>Jenkins is currently running tests in the <code>sf</code> test suite.</p><h4>Base Branch:</h4><p>Branch Name: <a href="https://github.com/liferay/liferay-portal/tree/master">master</a></p><h4>Job Summary:</h4><p>Job Link: <a href="https://test-1-6.liferay.com/job/test-portal-source-format/5724/">test-portal-source-format</a></p><p><b>For more details click <a href="https://test-1-6.liferay.com//userContent/jobs/test-portal-source-format/builds/5724/jenkins-report.html">here</a>.</b></p>

liferay-continuous-integration commented:
<h1>Build started.</h1><p>Jenkins is currently running tests in the <code>relevant</code> test suite.</p><h4>Base Branch:</h4><p>Branch Name: <a href="https://github.com/liferay/liferay-portal/tree/master">master</a></p><h4>Job Summary:</h4><p>Job Link: <a href="https://test-1-26.liferay.com/job/test-portal-acceptance-pullrequest(master)/38/">test-portal-acceptance-pullrequest(master)</a></p><p><b>For more details click <a href="https://test-1-26.liferay.com//userContent/jobs/test-portal-acceptance-pullrequest(master)/builds/38/jenkins-report.html">here</a>.</b></p>

liferay-continuous-integration commented:
<html><h3>:heavy_check_mark: ci:test:sf - 1 out of 1 jobs passed in 4 minutes</h3><details><summary>Click here for more details.</summary><h4>Base Branch:</h4><p>Branch Name: <a href="https://github.com/liferay/liferay-portal/tree/master">master</a><br/>Branch GIT ID: <a href="https://github.com/liferay/liferay-portal/commit/4c3862d7b511114e8a6ad8768602f5c1cdb9ab89">4c3862d7b511114e8a6ad8768602f5c1cdb9ab89</a></p><h4>Sender Branch:</h4><p>Branch Name: <a href="https://github.com/kresimir-coko/liferay-portal/tree/LPS-138446">LPS-138446</a><br/>Branch GIT ID: <a href="https://github.com/kresimir-coko/liferay-portal/commit/9572876e5f14a8d9b83b32960370f8e939a5cd7c">9572876e5f14a8d9b83b32960370f8e939a5cd7c</a></p>1 out of 1jobs PASSED<details><summary><strong>1 Successful Jobs:</strong></summary><ul><li><a href="https://test-1-6.liferay.com/job/test-portal-source-format/5724/">test-portal-source-format</a></li></ul></details><h5>For more details click <a href="https://test-1-6.liferay.com/userContent/jobs/test-portal-source-format/builds/5724/jenkins-report.html">here</a>.</h5></details></html>

liferay-continuous-integration commented:
<div><div><strong>Jenkins Build:</strong><span></span><a href="https://test-1-6.liferay.com/job/test-portal-source-format/5724/">test-portal-source-format#5724</a><br/></div><div><strong>Jenkins Report:</strong><span></span><a href="https://test-1-6.liferay.com/userContent/jobs/test-portal-source-format/builds/5724/jenkins-report.html">jenkins-report.html</a><br/></div><div><strong>Jenkins Suite:</strong><span></span>sf<br/></div><div><strong>Pull Request:</strong><span></span><a href="https://github.com/liferay-frontend/liferay-portal/pull/1426">liferay-frontend#1426</a><br/></div><div><strong>Testray Routine:</strong><span></span><a href="https://testray.liferay.com/home/-/testray/builds?testrayRoutineId=1306753894">EE Pull Request</a><br/></div><div><strong>Testray Build:</strong><span></span><a href="https://testray.liferay.com/home/-/testray/runs?testrayBuildId=1557883552">[master] ci:test:sf - kresimir-coko &gt; liferay-frontend - PR#1426 - 2021-09-01[07:11:20]</a><br/></div><div><strong>Testray Importer:</strong><span></span><a href="https://test-1-6.liferay.com/job/publish-testray-report/1806/">publish-testray-report#1806</a><br/></div></div>

liferay-continuous-integration commented:
<html><h3>:heavy_check_mark: ci:test:stable - 10 out of 10 jobs passed</h3><h3>:x: ci:test:relevant - 18 out of 21 jobs passed in 3 hours 32 minutes</h3><details><summary>Click here for more details.</summary><h4>Base Branch:</h4><p>Branch Name: <a href="https://github.com/liferay/liferay-portal/tree/master">master</a><br/>Branch GIT ID: <a href="https://github.com/liferay/liferay-portal/commit/4c3862d7b511114e8a6ad8768602f5c1cdb9ab89">4c3862d7b511114e8a6ad8768602f5c1cdb9ab89</a></p><h4>Upstream Comparison:</h4><p>Branch GIT ID: <a href="https://github.com/liferay/liferay-portal/commit/4c3862d7b511114e8a6ad8768602f5c1cdb9ab89">4c3862d7b511114e8a6ad8768602f5c1cdb9ab89</a><br/>Jenkins Build URL: <a href="https://test-1-1.liferay.com/job/test-portal-acceptance-upstream-dxp(master)//2276">Acceptance Upstream DXP (master) #2276</a></p><details><summary><strong>ci:test:stable - 10 out of 10 jobs PASSED</strong></summary><details><summary><strong>10 Successful Jobs:</strong></summary><ul><li><a href="https://test-1-28.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/696/">test-portal-acceptance-pullrequest-batch(master)/functional-smoke-tomcat90-mysql57-jdk8_stable/0</a></li><li><a href="https://test-1-1.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/426513/">test-portal-acceptance-pullrequest-batch(master)/functional-upgrade-tomcat90-mysql57-jdk8_stable/0</a></li><li><a href="https://test-1-28.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/699/">test-portal-acceptance-pullrequest-batch(master)/modules-compile-jdk8/0</a></li><li><a href="https://test-1-1.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/426514/">test-portal-acceptance-pullrequest-batch(master)/modules-integration-mysql57-jdk8_stable/0</a></li><li><a href="https://test-1-1.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/426511/">test-portal-acceptance-pullrequest-batch(master)/modules-unit-jdk8_stable/0</a></li><li><a href="https://test-1-2.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/16817/">test-portal-acceptance-pullrequest-batch(master)/pmd-jdk8/0</a></li><li><a href="https://test-1-15.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/343883/">test-portal-acceptance-pullrequest-batch(master)/portal-license-smoke-jdk8/0</a></li><li><a href="https://test-1-1.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/426512/">test-portal-acceptance-pullrequest-batch(master)/rest-builder-jdk8/0</a></li><li><a href="https://test-1-28.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/697/">test-portal-acceptance-pullrequest-batch(master)/service-builder-jdk8/0</a></li><li><a href="https://test-1-28.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/700/">test-portal-acceptance-pullrequest-batch(master)/unit-jdk8_stable/0</a></li></ul></details></details><details><summary><strong>ci:test:relevant - 18 out of 21 jobs PASSED</strong></summary><div><h4>3 Failed Jobs:</h4><ul><li><a href="https://test-1-26.liferay.com/job/test-portal-acceptance-pullrequest(master)/38/"><strike><strong>test-portal-acceptance-pullrequest(master)</strong></strike></a></li><li><a href="https://test-1-30.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/1864/"><strike><strong>test-portal-acceptance-pullrequest-batch(master)/modules-integration-mysql57-jdk8/0</strong></strike></a></li><li><a href="https://test-1-18.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/234239/"><strike><strong>test-portal-acceptance-pullrequest-batch(master)/semantic-versioning-jdk8/0</strong></strike></a></li></ul></div><details><summary><strong>18 Successful Jobs:</strong></summary><ul><li><a href="https://test-1-15.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/343885/">test-portal-acceptance-pullrequest-batch(master)/central-requirements-jdk8/0</a></li><li><a href="https://test-1-15.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/343884/">test-portal-acceptance-pullrequest-batch(master)/functional-smoke-tomcat90-mysql57-jdk11_open/0</a></li><li><a href="https://test-1-28.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/696/">test-portal-acceptance-pullrequest-batch(master)/functional-smoke-tomcat90-mysql57-jdk8_stable/0</a></li><li><a href="https://test-1-28.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/702/">test-portal-acceptance-pullrequest-batch(master)/functional-upgrade-tomcat90-mysql57-jdk8/0</a></li><li><a href="https://test-1-1.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/426513/">test-portal-acceptance-pullrequest-batch(master)/functional-upgrade-tomcat90-mysql57-jdk8_stable/0</a></li><li><a href="https://test-1-15.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/343882/">test-portal-acceptance-pullrequest-batch(master)/lpkg-base-jdk8/0</a></li><li><a href="https://test-1-28.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/701/">test-portal-acceptance-pullrequest-batch(master)/lpkg-controller-jdk8/0</a></li><li><a href="https://test-1-28.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/699/">test-portal-acceptance-pullrequest-batch(master)/modules-compile-jdk8/0</a></li><li><a href="https://test-1-1.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/426514/">test-portal-acceptance-pullrequest-batch(master)/modules-integration-mysql57-jdk8_stable/0</a></li><li><a href="https://test-1-18.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/234238/">test-portal-acceptance-pullrequest-batch(master)/modules-unit-jdk8/0</a></li><li><a href="https://test-1-1.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/426511/">test-portal-acceptance-pullrequest-batch(master)/modules-unit-jdk8_stable/0</a></li><li><a href="https://test-1-28.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/698/">test-portal-acceptance-pullrequest-batch(master)/modules-unit-project-templates-jdk8/0</a></li><li><a href="https://test-1-2.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/16817/">test-portal-acceptance-pullrequest-batch(master)/pmd-jdk8/0</a></li><li><a href="https://test-1-15.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/343883/">test-portal-acceptance-pullrequest-batch(master)/portal-license-smoke-jdk8/0</a></li><li><a href="https://test-1-1.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/426512/">test-portal-acceptance-pullrequest-batch(master)/rest-builder-jdk8/0</a></li><li><a href="https://test-1-28.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/697/">test-portal-acceptance-pullrequest-batch(master)/service-builder-jdk8/0</a></li><li><a href="https://test-1-28.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/695/">test-portal-acceptance-pullrequest-batch(master)/unit-jdk8/0</a></li><li><a href="https://test-1-28.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/700/">test-portal-acceptance-pullrequest-batch(master)/unit-jdk8_stable/0</a></li></ul></details></details><h5>For more details click <a href="https://test-1-26.liferay.com/userContent/jobs/test-portal-acceptance-pullrequest(master)/builds/38/jenkins-report.html">here</a>.</h5><hr/><h4>Failures unique to this pull:</h4><ol><li><div><h5><a href="https://test-1-18.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/234239/">test-portal-acceptance-pullrequest-batch(master)/semantic-versioning-jdk8/0</a></h5><div><h6>Job Results:</h6><p>0 Tests Passed.<br/>1 Test Failed.</p></div><ol><li><div><a href="https://test-1-18.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/AXIS_VARIABLE=0/234239//consoleText">AXIS_VARIABLE=0 #234239</a><div><p>Please fix <strong>semantic versioning</strong> on <strong><a href="https://github.com/kresimir-coko/liferay-portal/tree/LPS-138446">kresimir-coko/LPS-138446</a></strong><pre><code>     [exec] 		+   method     cloneWithOriginalValues()
     [exec] 			+   return     com.liferay.expando.kernel.model.ExpandoColumn
     [exec] 	&lt;   class      com.liferay.portlet.expando.model.impl.ExpandoRowBaseImpl
     [exec] 		+   method     cloneWithOriginalValues()
     [exec] 			+   return     com.liferay.expando.kernel.model.ExpandoRow
     [exec] 	&lt;   class      com.liferay.portlet.expando.model.impl.ExpandoRowImpl
     [exec] 		+   method     cloneWithOriginalValues()
     [exec] 			+   return     com.liferay.expando.kernel.model.ExpandoRow
     [exec] 	&lt;   class      com.liferay.portlet.expando.model.impl.ExpandoRowModelImpl
     [exec] 		+   method     cloneWithOriginalValues()
     [exec] 			+   return     com.liferay.expando.kernel.model.ExpandoRow
     [exec] 	&lt;   class      com.liferay.portlet.expando.model.impl.ExpandoTableBaseImpl
     [exec] 		+   method     cloneWithOriginalValues()
     [exec] 			+   return     com.liferay.expando.kernel.model.ExpandoTable
     [exec] 	&lt;   class      com.liferay.portlet.expando.model.impl.ExpandoTableImpl
     [exec] 		+   method     cloneWithOriginalValues()
     [exec] 			+   return     com.liferay.expando.kernel.model.ExpandoTable
     [exec] 	&lt;   class      com.liferay.portlet.expando.model.impl.ExpandoTableModelImpl
     [exec] 		+   method     cloneWithOriginalValues()
     [exec] 			+   return     com.liferay.expando.kernel.model.ExpandoTable
     [exec] 	&lt;   class      com.liferay.portlet.expando.model.impl.ExpandoValueBaseImpl
     [exec] 		+   method     cloneWithOriginalValues()
     [exec] 			+   return     com.liferay.expando.kernel.model.ExpandoValue
     [exec] 	&lt;   class      com.liferay.portlet.expando.model.impl.ExpandoValueImpl
     [exec] 		+   method     cloneWithOriginalValues()
     [exec] 			+   return     com.liferay.expando.kernel.model.ExpandoValue
     [exec] 	&lt;   class      com.liferay.portlet.expando.model.impl.ExpandoValueModelImpl
     [exec] 		+   method     cloneWithOriginalValues()
     [exec] 			+   return     com.liferay.expando.kernel.model.ExpandoValue
     [exec] [Baseline Report] Mode: diff (persisted)
     [exec] Semantic versioning is incorrect while checking /opt/dev/projects/github/liferay-portal/portal-impl/portal-impl.jar against /opt/dev/projects/github/liferay-portal/.gradle/caches/modules-2/files-2.1/com.liferay.portal/com.liferay.portal.impl/7.8.8/19163901cec0d556aa747a325ea28da69deb2732/com.liferay.portal.impl-7.8.8.jar</code></pre></p></div></div></li></ol></div></li><li><div><h5><a href="https://test-1-30.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/1864/">test-portal-acceptance-pullrequest-batch(master)/modules-integration-mysql57-jdk8/0</a></h5><div><h6>Job Results:</h6><p>391 Tests Passed.<br/>1 Test Failed.</p></div><ol><li><div><a href="https://test-1-30.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/AXIS_VARIABLE=0/1864//consoleText">AXIS_VARIABLE=0 #1864</a><ol><li><div><a href="https://test-1-30.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/AXIS_VARIABLE=0/1864//testReport/com.liferay.portal.log.assertor/PortalLogAssertorTest/testScanXMLLog">com.liferay.portal.log.assertor.PortalLogAssertorTest.testScanXMLLog</a><pre><code>junit.framework.AssertionFailedError: 
com.liferay.portal.verify.VerifyException: com.liferay.commerce.currency.exception.NoSuchCurrencyException: No CommerceCurrency exists with the key {companyId=20100, code=USD}
com.liferay.portal.verify.VerifyException: com.liferay.commerce.currency.exception.NoSuchCurrencyException: No CommerceCurrency exists with the key {companyId=20100, code=USD}
	at com.liferay.portal.verify.VerifyProcess.verify(VerifyProcess.java:85)
	at com.liferay.portal.verify.extender.internal.osgi.commands.VerifyProcessTrackerOSGiCommands.executeVerifyProcesses(VerifyProcessTrackerOSGiCommands.java:293)
	at com.liferay.portal.verify.extender.internal.osgi.commands.VerifyProcessTrackerOSGiCommands$1.run(VerifyProcessTrackerOSGiCommands.java:345)
	at com.liferay.portal.output.stream.container.internal.OutputStreamContainerFactoryTrackerImpl.runWithSwappedLog(OutputStreamContainerFactoryTrackerImpl.java:134)
	at com.liferay.portal.verify.extender.internal.osgi.commands.VerifyProcessTrackerOSGiCommands.executeVerifyProcesses(VerifyProcessTrackerOSGiCommands.java:340)
	at com.liferay.portal.verify.extender.internal.osgi.commands.VerifyProcessTrackerOSGiCommands._execute(VerifyProcessTrackerOSGiCommands.java:394)
	at com.liferay.portal.verify.extender.internal.osgi.commands.VerifyProcessTrackerOSGiCommands.access$200(VerifyProcessTrackerOSGiCommands.java:79)
	at com.liferay.portal.verify.extender.internal.osgi.commands.VerifyProcessTrackerOSGiCommands$VerifyServic...</code></pre></div></li></ol></div></li></ol></div></li></ol><h4>For upstream results, click <a href="https://test-1-1.liferay.com/job/test-portal-acceptance-upstream-dxp(master)">here</a>.</h4></details></html><h5>Test bundle downloads:</h5><ul><li><a href="https://files.liferay.com/private/bundles/test-1-26/jobs/test-portal-acceptance-pullrequest(master)/builds/38/liferay-portal-bundle-tomcat.tar.gz">liferay-portal-bundle-tomcat.tar.gz</a> (mirrors: <a href="http://mirrors.dlc.liferay.com/files.liferay.com/private/bundles/test-1-26/jobs/test-portal-acceptance-pullrequest(master)/builds/38/liferay-portal-bundle-tomcat.tar.gz">dlc</a>, <a href="http://mirrors.lax.liferay.com/files.liferay.com/private/bundles/test-1-26/jobs/test-portal-acceptance-pullrequest(master)/builds/38/liferay-portal-bundle-tomcat.tar.gz">lax</a>)</li></ul>

liferay-continuous-integration commented:
<div><div><strong>Jenkins Build:</strong><span></span><a href="https://test-1-26.liferay.com/job/test-portal-acceptance-pullrequest(master)/38/">test-portal-acceptance-pullrequest(master)#38</a><br/></div><div><strong>Jenkins Report:</strong><span></span><a href="https://test-1-26.liferay.com/userContent/jobs/test-portal-acceptance-pullrequest(master)/builds/38/jenkins-report.html">jenkins-report.html</a><br/></div><div><strong>Jenkins Suite:</strong><span></span>relevant<br/></div><div><strong>Pull Request:</strong><span></span><a href="https://github.com/liferay-frontend/liferay-portal/pull/1426">liferay-frontend#1426</a><br/></div><div><strong>Testray Routine:</strong><span></span><a href="https://testray.liferay.com/home/-/testray/builds?testrayRoutineId=1306753894">EE Pull Request</a><br/></div><div><strong>Testray Build:</strong><span></span><a href="https://testray.liferay.com/home/-/testray/runs?testrayBuildId=1557930224">[master] ci:test:relevant - kresimir-coko &gt; liferay-frontend - PR#1426 - 2021-09-01[07:10:57]</a><br/></div><div><strong>Testray Importer:</strong><span></span><a href="https://test-1-26.liferay.com/job/publish-testray-report/105/">publish-testray-report#105</a><br/></div></div>

bryceosterhaus commented:
I agree with @carloslancha, we shouldn't commit these without the actual rules existing in DXP yet. I would imagine a commit history looking like

1. update npm-scripts to the version that contains the newest eslint-config
2. disable all new breaking rules (no-aui, new-line-siblings)
3. enable rule-A
4. fix all rule-A errors
5. enable rule-B
6. fix all rule-B errors
7. ... etc

kresimir-coko commented:
I've added a commit that uses the newest version of `npm-scripts` which introduces these rules. @carloslancha it should be good to review now.

carloslancha commented:
Nice, let's go with the first rule ;)

kresimir-coko commented:
@carloslancha here it should be, `ant all` works fine with the new version now. Should I bundle this with code that fixes the first version, or should this be a standalone, as the [task](https://issues.liferay.com/browse/LPS-138446) suggests?

carloslancha commented:
I'd bundle it, just to create a pr with a commit history as @bryceosterhaus suggested

kresimir-coko commented:
Yeah, gonna return this to draft, just so it doesn't get merged. But we also need to decide what to do with the [empty-line-between-elements](https://github.com/liferay/liferay-frontend-projects/pull/622) rule. With this version of `npm-scripts` it is in DXP and will cause massive changes, as it autofixes those problems, and there's 709 errors atm.

/cc @bryceosterhaus what do we do about the `empty-line-between-elements` being in DXP now as well?

kresimir-coko commented:
I've added 3 commits:
- https://github.com/liferay-frontend/liferay-portal/pull/1426/commits/6e93a6666c011d7cd9c7bd0681f26dd77c4bf1ed enables the `no-get-body` rule, I chose this one because it has only 16 occurrences and is pretty simple to fix, so something to get the ball rolling
- https://github.com/liferay-frontend/liferay-portal/pull/1426/commits/1f4b2c1bf343f7b302f7d43af3b7dbc5e88b673c from my limited knowledge of how it worked before, it works the same now. I tested it by manually changing the locale and testing if the inputs are moving the cursor to the right side.
- https://github.com/liferay-frontend/liferay-portal/pull/1426/commits/33fd75428f418dbe786399302c017dfafa2ff4e2 this one is untested, even though it is an obvious change. I tried just finding any editor in DXP and testing this, but the file didn't trigger

bryceosterhaus commented:
> what do we do about the empty-line-between-elements being in DXP now as well?

I would try enabling it and use the auto-fix and check out all the occurrences it fixes. If they all seem reasonable at the code level, go ahead and test it manually and then we can add another commit for that like the other AUI rules

bryceosterhaus commented on modules/apps/frontend-js/frontend-js-web/src/main/resources/META-INF/resources/liferay/events.js:
```diff
@@ -35,18 +35,26 @@
-					Liferay._baseDelegate,
-					'.lfr-delegate-click'
-				);
+				document.body.addEventListener('click', (event) => {
+					const delegateTarget = event.target.closest(
+						'.lfr-delegate-click'
+					);
+
+					if (delegateTarget) {
+						event.delegateTarget = delegateTarget;
+
+						Liferay._baseDelegate(event);
+					}
+				});
+
+				Liferay._baseDelegateHandle = true;
```
I think we should use this from dxp, [delegate](https://github.com/liferay/liferay-portal/blob/0849b75262d3a8c2277a5e10a9d822160d602ff7/modules/apps/frontend-js/frontend-js-web/src/main/resources/META-INF/resources/liferay/delegate/delegate.es.js#L45)

kresimir-coko replied:
I wasn't sure how to import it there. When I added it to the top, it threw a lot of errors in the console. So I just went with the regular approach to not spend a lot of time trying to find an example of it working.

bryceosterhaus replied:
Yeah I'm not actually too sure how to add that in here appropriately...

I think it might just be best to add an ignore comment for this instance of aui-body and leave the original delegate logic.

Since this logic is fairly core to DXP, I wouldn't want to accidentally cause a regression here.

kresimir-coko commented:
@bryceosterhaus I've added a couple new commits, the https://github.com/liferay-frontend/liferay-portal/pull/1426/commits/61df81c941a40f3aa9983ff43037850cbe559517 commit is another one that could use the `delegate` utility.

I think that if we were to replace the utilities with our own, we'd have to add them to [global scope](https://github.com/liferay/liferay-portal/blob/master/modules/apps/frontend-js/frontend-js-web/src/main/resources/META-INF/resources/liferay/global.es.js), which as far as I know is something we don't want to do.

There are 8 errors left, with 7 of them being inside deprecated files. Not sure what to do about those 7, but the [remaining 1](https://github.com/liferay/liferay-portal/blob/master/modules/apps/frontend-js/frontend-js-web/src/main/resources/META-INF/resources/liferay/portlet.js#L80) I will work on tomorrow, having a difficult time locating it in the UI atm.

bryceosterhaus commented on modules/apps/layout/layout-admin-web/src/main/resources/META-INF/resources/js/layout_customization_settings.js:
```diff
@@ -181,14 +179,19 @@ AUI.add(
-							'.layout-customizable-checkbox',
-							instance
-						);
+						document.body.addEventListener(EVENT_CLICK, (event) => {
```
Same with this delegate. Let's just use an eslint-ignore for the specific no-body rule

bryceosterhaus reviewed (commented):
> I think that if we were to replace the utilities with our own, we'd have to add them to global scope, which as far as I know is something we don't want to do.

Yeah, ideally we don't want to do that, I noted for the specific instances but I think its probably best to just leave the delegate code and ignore the eslint rule for that specific case of getBody.

For the deprecated files, I would say we can also just as eslint-ignore rules there. I'd rather preserve the functionality and suppress the warning in case we accidentally break the logic of those files.

kresimir-coko commented:
Hey @bryceosterhaus I've added the ignore line to the delegate and deprecated usages. I've also rebased and force-pushed.

bryceosterhaus commented:
Looking good. I would say continue to keep it as a draft and keep working on enabling the rules one-by-one. I don't think there is a rush to get this merged, so we can probably wait and include it all in this same PR once the rules are all on and code updated.

kresimir-coko commented:
Sure @bryceosterhaus, I am worried about it becoming a massive PR, so I'll split them into branches in case we decide to split it into multiple PRs too.

bryceosterhaus commented:
@kresimir-coko thats totally fine too. We can merge them one by one, but lets at least wait for this GA and then start merging

kresimir-coko commented:
Hey @bryceosterhaus I've rebased my changes, squashed them, and combined them with your https://github.com/liferay-frontend/liferay-portal/pull/1540 PR. I ran both `yarn checkFormat` and `ant format-source-current-branch` to make sure it works locally but I couldn't fire up a server as I'm having `ant all` issues.

bryceosterhaus commented:
@kresimir-coko it looks like you still need to rebase this branch off of upstream, as I think the only commit not in upstream is your `LPS-138446 replace of ignore usages of A.getBody()` commit

bryceosterhaus commented:
@kresimir-coko can you also give me push access to your liferay-portal repo? That way I can also rebase and push to your branch in case we aren't online at the same time(which is very likely).

bryceosterhaus commented on modules/apps/frontend-js/frontend-js-aui-web/src/main/resources/META-INF/resources/liferay/layout.js:
```diff
@@ -436,28 +434,6 @@ AUI.add(
 
 			var eventHandles = [];
 
-			if (A.UA.ie || A.UA.edge) {
-				eventHandles.push(
-					BODY.delegate(
-						'mouseenter',
-						(event) => {
-							event.currentTarget.addClass('focus');
-						},
-						'.portlet'
-					)
-				);
-
-				eventHandles.push(
-					BODY.delegate(
-						'mouseleave',
-						(event) => {
-							event.currentTarget.removeClass('focus');
-						},
-						'.portlet'
-					)
-				);
-			}
```
@kresimir-coko did we replace this somewhere? I would expect this to cause issues since we are removing a delegate that adds a class

kresimir-coko commented:
I've force-pushed again @bryceosterhaus
